docs/reference/modules/node.asciidoc

```diff
@@ -1,32 +1,223 @@
+is used only from external requests.
+
+Each node in the cluster serves one or more purposes, and there are
+four types of nodes:
+
+[cols="<,<,<,<",options="header",]
+|=======================================================================
+|Node Type |Setting |Default |Description
+|<<master,Master>> |`node.master` |`true` |A node that is eligible to control the cluster.
+|<<data,Data>> |`node.data` |`true` |A node that stores index data for search activity.
+|<<client,Client>> |`node.client` |`false` |A standalone node that routes requests, but
+does not hold data or handle master duties.
+|<<modules-tribe,Tribe>> |`tribe.*` |None |A special type of client node
+that allows you to connect to multiple clusters.
+|=======================================================================
+
+By default, if you do not touch any settings, then a node is both a
+master eligible node and a data node. This is extremely convenient for
+small clusters, such as if you create a test cluster locally on your own
+computer for development.
+
+NOTE: A master node is said to be _eligible_ because there is a
+<<modules-discovery-zen,master election process>> that determines which
+master node (if there is more than one) is elected.
+
+For larger clusters, it generally makes sense to have standalone data nodes
+and standalone master nodes. To do this, you must modify the configuration
+of the respective nodes.
+
+[float]
+[[master]]
+===== Master Eligible Node
+
+Creating a standalone master node has many benefits, but the most obvious
+is that it offloads work from another node. Master nodes control the
+cluster by determining where shards will exist, as well as determining
+what nodes are (and therefore are not) connected to the cluster.
+
+Master nodes _can_ receive search and indexing traffic from clients
+(e.g., you) and they will properly route the request, as will any other
+node, but you should never send these types of requests to a 
+standalone master node; standalone master nodes will inherently
+_never_ have data, which means that the request is an unnecessary
+burden. An easy way to think about it is: you are tasking the node to
+do work that is not part of its normal duties.
+
+To create a standalone master eligible node:
+
+```yaml
+node.master: true
+node.data: false
+```
+
+NOTE: It is not enough to just set `node.master: true`. You must
+disable the data aspect, which is enabled by default.
+
+[float]
+[[data]]
+===== Data Node
+
+Creating a standalone data node has the same benefits as creating a
+standalone master node: offload unnecessary workload. By removing
+master duties from a data node, you free up threads to do what you
+need them to do best: search and index data.
+
+Just like any other node, data nodes can handle requests for
+search and indexing traffic, as well as cluster related traffic
+(e.g., requesting the current cluster settings). Data nodes are the
+nodes that literally contain the data, so they are meant to handle
```

Comment:
I'd delete this sentence.  Cluster level actions are pretty low key, it's fine for them to be sent to the data nodes.  Better than trying to maintain separate node pools for different actions in your client.  Rather just send all requests to non-master-eligible nodes.